plugin.rb

```diff
@@ -60,13 +62,27 @@
 
   add_to_class(:guardian, :can_assign?) { user && user.can_assign? }
 
+  add_to_serializer(:site, :assign_allowed_on_groups) do
+    staff_group = Group.select(:name).find_by(id:Group::AUTO_GROUPS[:staff])
```

Comment:
I am so uneasy having this query every we load the page. 

Maybe instead this site setting just had ids in it?

https://github.com/discourse/discourse/blob/5bc5c02af6877d0aed5ccdadf7bcf5f70385887e/app/models/group.rb#L70-L70

Staff has the group id 3, so you can start with that as the default. A bunch of stuff cleans up internally once that is in play.